v2/webui.go

```diff
@@ -268,19 +275,39 @@ func Wait() {
 }
 
 // Bind a specific html element click event with a function. Empty element means all events.
-func Bind(window uint, element string, callback func(Event) string) {
+func (w Window) Bind(element string, callback func(Event) any) {
 	iniModule()
 
 	// Convert element from Go-String to C-String
 	c_element := C.CString(element)
-	C.go_webui_bind(C.size_t(window), c_element)
+	C.go_webui_bind(C.size_t(w), c_element)
 
 	// Get the window ID
-	var window_id uint = uint(C.webui_interface_get_window_id(C.size_t(window)))
+	var window_id uint = uint(C.webui_interface_get_window_id(C.size_t(w)))
 
 	// Generate a unique ID for this element
 	var func_id string = strconv.Itoa(int(window_id)) + element
 
 	// Add the user callback function to the list
 	fun_list[func_id] = callback
 }
+
+func (d Data) String() string {
+	return string(d)
+}
+
+func (d Data) Int() int {
+	num, err := strconv.Atoi(string(d))
+	if err != nil {
+		log.Println("Failed getting event int argument", err)
+	}
+	return num
+}
+
+func (d Data) Bool() bool {
+	boolVal, err := strconv.ParseBool(string(d))
+	if err != nil {
+		log.Println("Failed getting event bool argument", err)
+	}
+	return boolVal
+}
```

Comment:
Not implementing the Data decode methods, but waiting until the webui_return_<typ> functions are implanted would also be okay I think. The methods would just add a little convenience so users wouldn't have to use strconv. on e.Data themselves.

I'm just logging, instead of returning errors in this quickfix, as `webui_return_<typ>` would also not return errors.

Reply:
I guess it's fine to implement these approaches.